core/src/main/java/org/apache/iceberg/rest/RESTSerializers.java

```diff
@@ -470,4 +494,112 @@ public T deserialize(JsonParser p, DeserializationContext context) throws IOExce
       return (T) LoadCredentialsResponseParser.fromJson(jsonNode);
     }
   }
+
+  static class PlanTableScanRequestSerializer<T extends PlanTableScanRequest>
+      extends JsonSerializer<T> {
+    @Override
+    public void serialize(T request, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      PlanTableScanRequestParser.toJson(request, gen);
+    }
+  }
+
+  static class PlanTableScanRequestDeserializer<T extends PlanTableScanRequest>
+      extends JsonDeserializer<T> {
+    @Override
+    public T deserialize(JsonParser p, DeserializationContext context) throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      return (T) PlanTableScanRequestParser.fromJson(jsonNode);
+    }
+  }
+
+  static class FetchScanTasksRequestSerializer<T extends FetchScanTasksRequest>
+      extends JsonSerializer<T> {
+    @Override
+    public void serialize(T request, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      FetchScanTasksRequestParser.toJson(request, gen);
+    }
+  }
+
+  static class FetchScanTasksRequestDeserializer<T extends FetchScanTasksRequest>
+      extends JsonDeserializer<T> {
+    @Override
+    public T deserialize(JsonParser p, DeserializationContext context) throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      return (T) FetchScanTasksRequestParser.fromJson(jsonNode);
+    }
+  }
+
+  static class PlanTableScanResponseSerializer<T extends PlanTableScanResponse>
+      extends JsonSerializer<T> {
+    @Override
+    public void serialize(T response, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      PlanTableScanResponseParser.toJson(response, gen);
+    }
+  }
+
+  static class PlanTableScanResponseDeserializer<T extends PlanTableScanResponse>
+      extends JsonDeserializer<T> {
+    @Override
+    public T deserialize(JsonParser p, DeserializationContext context) throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      // Retrieve injectable values
+      @SuppressWarnings("unchecked")
+      Map<Integer, PartitionSpec> specsById =
+          (Map<Integer, PartitionSpec>) context.findInjectableValue("specsById", null, null);
+
+      boolean caseSensitive = (boolean) context.findInjectableValue("caseSensitive", null, null);
```

Comment:
Minor: Maybe worth having a static class `TableScanResponseContext` with these two fields and we have a helper function ````parseScanResponseContext(DeserializationContext context) {return new TableScanResponseContext(context.find...))```